clients/cobol-lsp-vscode-extension/src/commands/RunAnalysisCLI.ts

```diff
@@ -0,0 +1,251 @@
+        this.buildAnalysisCommandPortion(currentFileLocation)
+      );
+    }
+
+    return "";
+  }
+
+  /**
+   * Provides the portion of the command from "analysis" onwards.
+   * Is the same for both the Java and Native builds.
+   * @param currentFileLocation - Location of the main cobol file being analyzed.
+   * @protected
+   */
+  protected buildAnalysisCommandPortion(currentFileLocation: string) {
+    const copyBookCommand = `-cf=${
+      this.copybookConfigLocation === ""
+        ? "."
+        : '"' + this.copybookConfigLocation + '"'
+    }`;
+
+    return 'analysis -s "' + currentFileLocation + '" ' + copyBookCommand;
+  }
+
+  /**
+   * Sends a given command to a terminal.
+   * Checks to see if one named "Analysis" is already created, if so clear and reuse it.
+   * @param command - The command to run from the terminal.
+   * @protected
+   */
+  protected sendToTerminal(command: string) {
+    const existingTerminal = vscode.window.terminals.find(
+      (term: Terminal) => term.name === "Analysis",
+    );
+    const terminal = existingTerminal
+      ? existingTerminal
+      : vscode.window.createTerminal("Analysis");
+
+    terminal.sendText(command);
+    terminal.show(true);
+  }
+
+  /**
+   * Retrieves the location of the currently open file.
+   * If it is not saved, creates a temporary one on the disk to send to the command line with a copy of the contents.
+   *    - Does not result in the actual file being saved to the end user.
+   * @protected
+   */
+  protected async getCurrentFileLocation() {
+    if (vscode.window.activeTextEditor) {
+      if (vscode.window.activeTextEditor.document.uri.scheme === "untitled") {
+        return this.saveTempFile();
+      } else {
+        return vscode.window.activeTextEditor.document.uri.fsPath;
+      }
+    }
+    return "";
+  }
+
+  /**
+   * Creates a temporary one on the disk to send to the command line with a copy of the contents.
+   *    - Does not result in the actual file being saved to the end user.
+   * @protected
+   */
+  protected async saveTempFile() {
+    const data = vscode.window.activeTextEditor?.document.getText();
+
+    if (data) {
+      const fileStatResult = await vscode.workspace.fs.stat(
+        this.globalStorageUri,
+      );
+      if (fileStatResult.type === vscode.FileType.Directory) {
+        const tempFolderUri = vscode.Uri.joinPath(
+          this.globalStorageUri,
+          "/tempAnalysisFiles",
+        );
+        try {
+          const tempFolderStatResult = await vscode.workspace.fs.stat(
+            tempFolderUri,
+          );
+          if (tempFolderStatResult.type === vscode.FileType.Directory) {
+            const tempFolderResult = await vscode.workspace.fs.readDirectory(
+              tempFolderUri,
+            );
+            tempFolderResult.map(async (value) => {
+              await vscode.workspace.fs.delete(
+                vscode.Uri.joinPath(tempFolderUri, value[0]),
+              );
+            });
```

Comment:
I suggest adding a comment here (the map generates an array full of promises that are not awaited).
```suggestion
            // deletions are intentionally not awaited
            tempFolderResult.map(async (value) => {
              await vscode.workspace.fs.delete(
                vscode.Uri.joinPath(tempFolderUri, value[0]),
              );
            });
```

Reply:
Adding `await` to the `tempFolderResult.map` results in a "TS80007: await has no effect on the type of this expression": 
![Screenshot 2024-07-05 at 7 51 56 AM](https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/assets/170667132/3e1df250-202c-4759-a014-323a3d8b882f)

This is due to `value` not being a promise itself, only that it is doing an async function call with value as the variable provided. The await within is the correct answer as that _will_ wait for deletions.

No need to worry about file name collisions as I am using `Date.now()` as the new temporary file name.

Reply:
I did not ask for the a top level await, I asked for a comment explaining why we are not awaiting the all promises (which we are not - the await in the map callback does not await the all the operations).

Reply:
JoinPath is not an async operation. The `delete` is properly awaited.

Reply:
This is not how it works. If you give the map function a name e.g.
```
const deleteFileEntry = async (value) => {
    await vscode.workspace.fs.delete(
        vscode.Uri.joinPath(tempFolderUri, value[0]),
    );
}
```
then your code is basically equivalent to the following set of calls
```
deleteFileEntry(tempFolderResult[0]);
deleteFileEntry(tempFolderResult[1]);
deleteFileEntry(tempFolderResult[2]);
deleteFileEntry(tempFolderResult[3]);
...
```
each producing a promise (that is not awaited), the map function just puts them all in a single array.

Reply:
Wrapped the map function with `Promise.all` in latest commit. Resolves the lack of an await and removes the error about the "no effect on this type of expression" warning.